canonical-web-and-design/maas-ui
# Only fetch once from sagas

## Done
1. Replace custom hook `useFetchOnce` with similar logic in `sendMessage` saga.
2. Add `model` meta field to actions, a string representing the corresponding root level object in redux state (e.g. users, repositories, config), so that dynamic selectors can operate over state in sagas.
3. `FETCH_*_LOADED` is now dispatched instead, which should ensure `loaded: true` and `loading: false` are set in state. `FETCH_*_LOADED` isn't _strictly_ necessary, but it felt like `FETCH_*` should have a corresponding response of some kind, otherwise it feels like the action was swallowed. Feel free to disagree!

## QA
With the redux inspector, or websocket network inspector open:
1. Visit the Users table
2. Follow a link to another view
3. Return to the Users table, expect to see:
  a. `FETCH_USERS_LOADED` dispatched
  b. No additional `FETCH_USERS_START` or `FETCH_USERS_SUCCESS` dispatched, nor any websocket requests for `users.list`.
